nats-base-client/kv.ts

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2021 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ConsumerOpts,
+  DiscardPolicy,
+  JetStreamClient,
+  JetStreamManager,
+  JetStreamPublishOptions,
+  JsMsg,
+  Nanos,
+  NatsConnection,
+  PurgeOpts,
+  PurgeResponse,
+  RetentionPolicy,
+  StorageType,
+  StoredMsg,
+  StreamConfig,
+} from "./types.ts";
+import { JetStreamClientImpl } from "./jsclient.ts";
+import { createInbox, headers, millis } from "./mod.ts";
+import { deferred, timeout } from "./util.ts";
+
+export type Result = {
+  bucket: string;
+  key: string;
+  data: Uint8Array; // data lines up better with other nats apis in javascript
+  created: Date;
+  sequence: number;
+  delta?: number;
+  "origin_cluster"?: string;
+  operation: "PUT" | "DEL"; // these are redundant - a delete is empty payload...
```

Comment:
not really redundant though is it, you might legit choose to put a empty payload in a key.